File: Hackerrank/pairs.java
/*
#5 2  
#1 5 3 4 2  
*/

import java.util.*;

class pairs
{
    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        int noe = scan.nextInt();
        int k   = scan.nextInt();
        
        int[] arr = new int[noe];
        for (int i=0; i < noe; i++)
        {
            arr[i] = scan.nextInt();
        }
        Arrays.sort(arr);
        int pair = 0;
        for (int i=0; i < noe; i++)
        {
            for (int j = i+1; j < noe; j++)
            {
                if (arr[j] - arr[i] == k)
                {
                    pair++;
                    break;
                }
                
            }
        }
        System.out.print(pair);
        
    }
}

File: Hackerrank/pairs.py
'''
5 2  
1 5 3 4 2  
'''

noe, k = [int(x) for x in input().split()]
arr = sorted([int(x) for x in input().split()])
pairs = 0
for i in range(noe):
    for j in range(i+1,noe):
        if arr[j] - arr[i] == k:
            pairs += 1
            break
print(pairs)        
